cmd/ocm/create/cluster/cmd.go

```diff
@@ -180,58 +182,104 @@ func run(cmd *cobra.Command, argv []string) error {
 	// Get the client for the cluster management api
 	cmv1Client := connection.ClustersMgmt().V1()
 
+	var clusterName string
 	// Check and set the cluster name
-	if len(argv) != 1 || argv[0] == "" {
-		return fmt.Errorf("A cluster name must be specified")
+	if len(argv) == 1 && argv[0] != "" {
+		clusterName = argv[0]
+	} else {
+		if args.interactive {
+			prompt := &survey.Input{
+				Message: "cluster name",
+				Help:    clusterNameHelp,
+			}
+			survey.AskOne(prompt, &clusterName, survey.WithValidator(survey.Required))
+		} else {
+			return fmt.Errorf("A cluster name must be specified")
+		}
+	}
+
+	// Validate flags / ask for missing data.
+
+	// Only offer the 2 providers known to support OSD now;
+	// but don't validate if set, to not block `ocm` CLI from creating clusters on future providers.
+	err = arguments.PromptOrCheckOneOf(cmd.Flags(), "provider", osdProviders())
+	if err != nil {
+		return err
+	}
+
+	err = arguments.PromptBool(cmd.Flags(), "ccs")
+	if err != nil {
+		return err
+	}
+	if args.ccs.Enabled {
+		err = arguments.PromptString(cmd.Flags(), "aws-access-key-id")
+		if err != nil {
+			return err
+		}
+
+		err = arguments.PromptPassword(cmd.Flags(), "aws-secret-access-key")
+		if err != nil {
+			return err
+		}
+	}
+	err = arguments.CheckIgnoredCCSFlags(args.ccs)
+	if err != nil {
+		return err
+	}
+
+	regionSet := sets.NewString()
+	regions, err := provider.GetRegions(connection.ClustersMgmt().V1(), args.provider, args.ccs)
+	if err != nil {
+		return err
+	}
+	for _, region := range regions {
+		regionSet.Insert(region.ID())
+	}
+	arguments.PromptOrCheckOneOf(cmd.Flags(), "region", regionSet.List())
+
+	if args.region == "us-east-1" && args.provider != "aws" {
+		return fmt.Errorf("if specifying a non-aws cloud provider, region must be set to a valid region")
 	}
-	clusterName := argv[0]
 
-	// Validate flags:
 	expiration, err := c.ValidateClusterExpiration(args.expirationTime, args.expirationSeconds)
 	if err != nil {
 		return fmt.Errorf(fmt.Sprintf("%s", err))
 	}
-	versionList := sets.NewString()
+
+	versionSet := sets.NewString()
 	versions, err := fetchEnabledVersions(cmv1Client)
 	if err != nil {
 		return fmt.Errorf("unable to retrieve versions: %s", err)
 	}
 	for _, version := range versions {
-		versionList.Insert(version.ID())
-
+		// strip leading "openshift-v" string
+		versionSet.Insert(strings.Replace(version.ID(), "openshift-v", "", 1))
 	}
+	arguments.PromptOrCheckOneOf(cmd.Flags(), "version", versionSet.List())
```

Comment:
Will the `default` version be always selected by default? Does `fetchEnabledVersions` do any sorting on that?

Reply:
Hmm, great question!  The flag has no default, normally default is applied server-side.
I should look at .Default field, right?

`fetchEnabledVersions` sorting won't matter as written as `versionSet.List()` does lexicographic sorting.
TODO: keep sorting but set default.
TODO: use "version sort" so that 2 < 11.